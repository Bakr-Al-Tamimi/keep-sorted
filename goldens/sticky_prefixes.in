Sort these lines with various comments:
Flags(
  // keep-sorted-test start sticky_prefixes=/**,@
  /** Flag to do important things */
  FLAG_OF_IMPORTANCE,
  @Deprecated
  DEPRECATED_FLAG,
  ALPHABETICALLY_FIRST_FLAG,
  // keep-sorted-test end
)

Multiple sticky prefixes per item:
// keep-sorted-test start sticky_prefixes=@1,@2
@1 zulu
@2 zulu
Zulu
@1 whiskey
Whiskey
@1 alpha
@2 alpha
Alpha
// keep-sorted-test end

With grouping:
public class Foo {
  // keep-sorted-test start group=yes sticky_prefixes=@1,@2
  private final Foo foo;
  private final Bar bar;
  @2 baz
  private final Baz baz =
      @1 new baz
      new Baz();
  private final Qux qux;
  // keep-sorted-test end
}

Without sticky comment:
// keep-sorted-test start sticky_comments=no sticky_prefixes=@1,@2
// Will go second.
@1 delta
Delta
Charlie
@1 bravo
@2 bravo
Bravo
// Always goes first
// keep-sorted-test end

Dangling sticky items:
// keep-sorted-test start sticky_prefixes=Sticky
Sticky Zulu
Zulu
Sticky Charlie
Sticky Bravo
Sticky Alpha
// keep-sorted-test end
