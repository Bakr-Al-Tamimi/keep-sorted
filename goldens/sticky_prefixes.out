Sort these lines with various comments:
Flags(
  // keep-sorted-test start sticky_prefixes=/**,@
  ALPHABETICALLY_FIRST_FLAG,
  @Deprecated
  DEPRECATED_FLAG,
  /** Flag to do important things */
  FLAG_OF_IMPORTANCE,
  // keep-sorted-test end
)

Multiple sticky prefixes per item:
// keep-sorted-test start sticky_prefixes=@1,@2
@1 alpha
@2 alpha
Alpha
@1 whiskey
Whiskey
@1 zulu
@2 zulu
Zulu
// keep-sorted-test end

With grouping:
public class Foo {
  // keep-sorted-test start group=yes sticky_prefixes=@1,@2
  private final Bar bar;
  @2 baz
  private final Baz baz =
      @1 new baz
      new Baz();
  private final Foo foo;
  private final Qux qux;
  // keep-sorted-test end
}

Without sticky comment:
// keep-sorted-test start sticky_comments=no sticky_prefixes=@1,@2
// Always goes first
// Will go second.
@1 bravo
@2 bravo
Bravo
Charlie
@1 delta
Delta
// keep-sorted-test end

Dangling sticky items:
// keep-sorted-test start sticky_prefixes=Sticky
Sticky Zulu
Zulu
Sticky Charlie
Sticky Bravo
Sticky Alpha
// keep-sorted-test end
